### PR TITLE
[FLINK-17303][python] Return TableResult for Python TableEnvironment

### PR DESCRIPTION
**Pull request by SteNicholas**
## What is the purpose of the change

*This pull request will introduce python `TableResult` class for python tableEnvironment to make sure consistent with Java, which could get `JobClient` (to associates the submitted Flink job), or print the execution result.*

## Brief change log

  - *Add `TableResult` class consistent with Java, including `get_job_client`, `get_table_schema`, `get_result_kind` and print methods for the representation of the statement execution result.*
  - *Add `JobClient` class that is scoped to a specific job, including `get_job_id`, `get_job_status`, `cancel`, `stop_with_savepoint`, `trigger_savepoint`, `get_accumulators` and `get_job_execution_result` for job execution result.*

## Verifying this change

  - *Add `test_execute_sql` method for `StreamSqlTests` to verify table result of DDL statement execution test case.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-python/pyflink/table/tests/test_sql.py [hunk 1]
```diff
@@ -58,6 +56,67 @@ def test_sql_query(self):
...
+        self.assertIsNotNone(table_result.get_job_client())
+        job_status_feature = table_result.get_job_client().get_job_status()
+        job_execution_result_feature = table_result.get_job_client().get_job_execution_result(
+            get_gateway().jvm.Thread.currentThread().getContextClassLoader())
+        job_execution_result = job_execution_result_feature.result()
+        self.assertIsNotNone(job_execution_result)
+        self.assertIsNotNone(job_execution_result.get_job_id())
+        self.assertIsNotNone(job_execution_result.get_job_execution_result())
+        job_status = job_status_feature.result()
+        self.assertIsNotNone(job_status)
```
Comment (hequn8128) [inline flink-python/pyflink/table/tests/test_sql.py:103]: Remove tests about job status. The cluster would exist before calling the `get_job_status` while leads to test failures.

File: flink-python/pyflink/table/tests/test_sql.py [hunk 2]
```diff
@@ -58,6 +56,67 @@ def test_sql_query(self):
...
+            "sinks",
+            source_sink_utils.TestAppendSink(field_names, field_types))
+        table_result = t_env.execute_sql("insert into sinks select * from tbl")
+        self.assertIsNotNone(table_result)
+        self.assertIsNotNone(table_result.get_job_client())
+        job_status_feature = table_result.get_job_client().get_job_status()
+        job_execution_result_feature = table_result.get_job_client().get_job_execution_result(
+            get_gateway().jvm.Thread.currentThread().getContextClassLoader())
+        job_execution_result = job_execution_result_feature.result()
+        self.assertIsNotNone(job_execution_result)
```
Comment (hequn8128) [inline flink-python/pyflink/table/tests/test_sql.py:99]: Remove this line.

**Comment by hequn8128** [comment]
@flinkbot run azure

**Comment by SteNicholas** [comment]
@flinkbot run azure

**Pull request closed without merging**